datalad/distribution/get.py

```diff
@@ -414,10 +415,21 @@ def _install_necessary_subdatasets(
     """
     # figuring out what dataset to start with, --contains limits --recursive
     # to visit only subdataset on the trajectory to the target path
-    subds_trail = ds.subdatasets(contains=path, recursive=True,
-                                 on_failure="ignore",
-                                 result_filter=is_ok_dataset,
-                                 result_renderer='disabled')
+    iterable = ds.subdatasets(
+        contains=path,
+        recursive=True,
+        on_failure='ignore',
+        result_filter=is_ok_dataset,
+        result_renderer='disabled',
+        return_type='generator')
+
+    subds_trail = yield from FilteringIterator(
+        iterable=iterable,
+        patterns=[{
+            'action': 'subdataset',
+            'type': 'dataset',
+            'status': 'ok'}])
+
```

Comment:
Dunno enough, but isn't it what `result_filter=is_ok_dataset` in prior call for?

Also isn't condition before would always be false since we would always have an operator? (On the phone ATM, can't see coverage)

Reply:
Good point. I should have removed "result_filter".

Result filter is different from FilteringIterator because the filtering iterator yields everything that is not filtered and keeps the filtered elements. A result filter, on the other hand, would not yield elements that do not pass the filter.